crates/store/re_log_encoding/src/rrd/encoder.rs

```diff
@@ -103,31 +103,86 @@ struct FooterState {
     /// want to override the recording ID of each chunk with that one (because that's the existing
     /// behavior, certainly not because it's nice).
     recording_id_scope: Option<re_log_types::StoreId>,
+
+    manifests: HashMap<re_log_types::StoreId, ManifestState>,
+}
+
+/// The accumulated state for a specific RRD manifest.
+#[derive(Default)]
+struct ManifestState {
+    /// The accumulated recording IDs of each individual chunk, extracted from their `LogMsg`.
+    ///
+    /// This will only be used if [`FooterState::recording_id_scope`] is empty.
+    recording_ids: Vec<re_log_types::StoreId>,
+
+    /// The state of the RRD manifest currently being built.
+    manifest: RrdManifestBuilder,
 }
 
 impl FooterState {
-    #[expect(clippy::unnecessary_wraps)] // won't stay for long
     fn append(
         &mut self,
-        _byte_offset: u64,
-        _byte_size: u64,
+        byte_offset: u64,
+        byte_size: u64,
```

Comment:
Can use `re_span::Span` here

```suggestion
        bytes: Span<u64>,
```